src/service/github-allocator-registry/github-allocator-registry.service.ts

```diff
@@ -0,0 +1,146 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { createAppAuth } from '@octokit/auth-app';
+import { Octokit } from '@octokit/core';
+import { ConfigService } from '@nestjs/config';
+import {
+  HealthCheckError,
+  HealthIndicator,
+  HealthIndicatorResult,
+} from '@nestjs/terminus';
+import { LotusApiService } from '../lotus-api/lotus-api.service';
+import { AllocatorRegistry } from './types.github-allocator-registry';
+
+@Injectable()
+export class GitHubAllocatorRegistryService extends HealthIndicator {
+  private readonly logger = new Logger(GitHubAllocatorRegistryService.name);
+  private healthy = true;
+  private octokit: Octokit;
+
+  constructor(
+    private readonly configService: ConfigService,
+    private readonly lotusApiService: LotusApiService,
+  ) {
+    super();
+  }
+
+  public async getHealth(): Promise<HealthIndicatorResult> {
+    const result = this.getStatus(
+      GitHubAllocatorRegistryService.name,
+      this.healthy,
+      {},
+    );
+
+    if (this.healthy) return result;
+    throw new HealthCheckError('Healthcheck failed', result);
+  }
+
+  private async getInstallationId(): Promise<number> {
+    const appOctokit = new Octokit({
+      authStrategy: createAppAuth,
+      auth: {
+        appId: this.configService.get<string>('GITHUB_APP_ID'),
+        privateKey: this.configService.get<string>('GITHUB_PRIVATE_KEY'),
+      },
+    });
+
+    const response = await appOctokit.request(
+      'GET /repos/{owner}/{repo}/installation',
+      {
+        owner: this.configService.get<string>('ALLOCATOR_REGISTRY_REPO_OWNER'),
+        repo: this.configService.get<string>('ALLOCATOR_REGISTRY_REPO_NAME'),
+        headers: {
+          'X-GitHub-Api-Version': '2022-11-28',
+        },
+      },
+    );
+
+    return response.data.id;
+  }
+
+  private async getOctokit(): Promise<Octokit> {
+    if (!this.octokit) {
+      this.octokit = new Octokit({
+        authStrategy: createAppAuth,
+        auth: {
+          appId: this.configService.get<string>('GITHUB_APP_ID'),
+          privateKey: this.configService.get<string>('GITHUB_PRIVATE_KEY'),
+          installationId: await this.getInstallationId(),
```

Comment:
I'm open for suggestions how to handle this double-octokit thing better.

Reply:
I can only think of extracting octokit creation to separate function to avoid code duplication. I'll push a commit to this branch, take a look and feel free to revert it if you don't like it.